nimbus/src/evaluator.rs

```diff
@@ -288,26 +313,33 @@ mod tests {
         };
         let mut experiment2 = experiment1.clone();
         experiment2.arguments.bucket_config = BucketConfig {
-            randomization_unit: RandomizationUnit::NormandyId,
+            randomization_unit: RandomizationUnit::ClientId,
             namespace:
                 "bug-1637316-message-aboutwelcome-pull-factor-reinforcement-76-rel-release-76-77"
                     .to_string(),
-            start: 2000,
-            count: 3000,
+            start: 9000,
+            count: 1000,
             total: 10000,
         };
         experiment2.arguments.slug = "TEST_EXP2".to_string();
         let experiments = vec![experiment1, experiment2];
+        let available_randomization_units = AvailableRandomizationUnits {
+            client_id: None, // We will not match EXP_2 because we don't have the necessary randomization unit.
+            normandy_id: None,
+        };
         // 299eed1e-be6d-457d-9e53-da7b1a03f10d uuid fits in start: 0, count: 2000, total: 10000 with the example namespace, to the treatment-variation-b branch
         // Tested against the desktop implementation
         let id = uuid::Uuid::parse_str("299eed1e-be6d-457d-9e53-da7b1a03f10d").unwrap();
-        let enrolled = filter_enrolled(&id, &experiments).unwrap();
+        let enrolled = filter_enrolled(&id, &available_randomization_units, &experiments).unwrap();
         assert_eq!(enrolled.len(), 1);
         assert_eq!(enrolled[0].slug, "TEST_EXP1");
-        // 542213c0-9aef-47eb-bc6b-3b8529736ba2 uuid fits in start: 2000, count: 3000, total: 10000 with the example namespace, to the control branch
-        // Tested against the desktop implementation
```

Comment:
The bucketing here *was not* tested against a reference implementation, do we have any tests that covers `normandy_id` or `client_id` that I could replicate?

Reply:
I don't think we have any test's here that replicate Normandy or Telemetry ids, but maybe something in the [nimbus-shared/test](https://github.com/mozilla/nimbus-shared/tree/main/test) repo covers what you are looking for?